cli/js/lib.deno.ns.d.ts

```diff
@@ -1266,6 +1268,18 @@ declare namespace Deno {
     isSymlink(): boolean;
   }
 
+  // File types (from st_mode & ~0o7777)
+  export enum FileType {
+    TYPE_UNKNOWN = 0, // whiteouts, doors, ports
+    TYPE_REGULAR = 8 << 12,
+    TYPE_DIRECTORY = 4 << 12,
+    TYPE_SYMLINK = 10 << 12,
+    TYPE_FIFO = 1 << 12,
+    TYPE_CHARDEV = 2 << 12,
+    TYPE_BLKDEV = 6 << 12,
+    TYPE_SOCKET = 12 << 12,
+  }
```

Comment:
These are platform dependent (?) and largely covered by `FileInfo.isDirectory()` , `isSymlink()`, `isFile()`... It's not clear to me it's desirable to expose such APIs.

Reply:
I can't speak authoritatively, but from the best I've been able to research, they don't vary (at least not between Linux and Mac).

I guess this PR does three things: (a) stop assuming that isDirectory, isSymlink, and isFile exhaust all the options (so don't define one of them as meaning "neither of the other two"); (b) expose a version of `st_mode` which is masked to have only the permission bits, so that we don't have to write `& 0o777` everytime we work with a `FileInfo.mode`; (c) expose the high bits of `st_mode` as an enum.

(c) concerns me the least, we could always add that back in later. Or I could create `isCharacterDevice` etc booleans on the `StatResponse` (in a way that uses the platform's own headers, in case these do vary). But I'm happy to take the enum out and leave this info unexposed, at least for now, if you think that's better.

I would still advocate for (b) though, and especially for (a).

Reply:
> at least not between Linux and Mac

It's windows I'm concerned about.

Reply:
I handled that by manually assigning the enum values when the StatResponse.mode field is null, based on the isFile etc fields.